### PR TITLE
Add optional secret argument for application creation

### PR DESCRIPTION
**Pull request by miks**
Make possible to create application with pregenerated password which is
very useful for automated deployments and tests where password is
securely generated and stored outside single service scope.

### CODE REVIEW

File: src/emqx_mgmt_cli.erl [hunk 1]
```diff
@@ -101,7 +111,7 @@ mgmt(["list"]) ->
 
 mgmt(_) ->
     emqx_cli:usage([{"mgmt list",                   "List Applications"},
-                    {"mgmt insert <AppId> <Name>",   "Add Application of REST API"},
+                    {"mgmt insert <AppId> <Name> [<Secret>]",   "Add Application of REST API"},
```
Comment (Gilbert-Wong) [inline src/emqx_mgmt_cli.erl]: What about `mgmt insert <AppId> <Name> <Secret>`

Comment (miks) [reply src/emqx_mgmt_cli.erl]: Other cli commands have use square brackets for optional arguments. Correct if I'm wrong.

Comment (Gilbert-Wong) [reply src/emqx_mgmt_cli.erl]: We have discussed the feature you implement, there are still controversy in our team. Anyway, appreciate your work and thanks for you contribution, we would give you feedback soon.

**Comment by Gilbert-Wong** [comment]
Sorry, @miks , Our team leader do not think it is good idea to create application with pregenerated password, maybe setting application id and pregenerated password in config is better. I have to close this pull request, anyway, really appreciate and thanks for your work.

**Comment by miks** [comment]
How it’s possible to set pregenerated password in config?

**Comment by Gilbert-Wong** [comment]
@miks , we have schedule to set pregenerated password in `emqx_management.conf`.

**Comment by alibo** [comment]
> @miks , we have schedule to set pregenerated password in emqx_management.conf.

@Gilbert-Wong Is there any ETA for this feature? 
Currently, I had to work on editing mnesia database during initialization (in [`/start.sh`](https://github.com/emqx/emqx-docker/blob/emqx30/start.sh)) which is a hacky solution! :(

**Comment by miks** [comment]
@Gilbert-Wong any updates on this?
As for https://github.com/emqx/emqx-docker it's little strange that I can pass EMQX_ADMIN_PASSWORD or EMQX_AUTH__REDIS__PASSWORD and there is no single problem with `Our team leader do not think it is good idea to create application with pregenerated password`.
As I stated before there is no use at the moment for EMQ in highly automated systems (CI, CD).
Maybe it's possible to review this decision to allow application creation with pregenerated password as other parts of EMQ have no problem at all that password(-s) is passed from external systems.

**Pull request closed without merging**